More API Hooks and Minor Windows Clipboard Module Fixes

jhumble opened the pull request:
Added some new API hooks and made some fixes to the clipboard module which was broken when last_error was moved from ql to ql.os.

xwings commented on qiling/os/windows/dlls/wsock32.py:
```diff
@@ -72,3 +72,29 @@ def hook_connect(ql, address, params):
+    "name": STRING
+})
+def hook_gethostbyname(ql, address, params):
+    dummy_ip = b"\x40\x30\x20\x10" #10.20.30.40
```
we should store this in profile? its in qiling/profile/windows.ql

jhumble replied:
Thanks! Do you want me to just add a [Network] section?

xwings replied:
sure, any appropriate name will work

jhumble replied:
Added as [NETWORK]: dns_response_ip

xwings commented on qiling/os/windows/dlls/kernel32/debugapi.py:
```diff
@@ -47,3 +47,13 @@ def hook_OutputDebugStringW(ql, address, params):
+    "lpOutputString": STRING
+})
+def hook_OutputDebugStringA(ql, address, params):
+    ql.os.stdout.write('OutputDebugStringA: "%s"' % (params['lpOutputString']))
```
will it be better to make this two lines of ql.os.stdout.write into ql.drpint ?

jhumble replied:
I'm not sure. I initially did have this implemented only as ql.dprint, but noticed that the existing OutputDebugStringW implementation only used ql.os.stdout.write, so I just changed mine to match it.

That doesn't seem correct though, since OutputDebugString will not actually emit any console input. I'll change both of them to use ql.dprint.

xwings replied:
the reson why ql.drpint or ql.nprint we are in control. we can choose to print or not to print. in this case should be nprint and not dprint.

jhumble replied:
Changed to nprint only

jhumble commented:
Cleaned up clipboard.py in accordance with the FIXME notes that were in it. No need for it to have a reference to ql, only to ql.os. 

Added profile.ql option to set the ip returned on dns requests (gethostbyname). 

Fixed a bug in the Mutex implementation in os/windows/structs.py (a variable and method were both named "lock"). 

Changed hook_GetLocaleInfoA and _CreateFile to both set the appropriate windows error on failure and return rather than crashing the emulator

Fixed other hooks that were throwing API Implementation Errors when running some bulk random executables through.

xwings commented:
Once again, good to have you back!

Thanks for the PR.
